src/modules/languages/rust.nix

```diff
@@ -80,8 +80,8 @@ in
     };
   };
 
-  config = lib.mkMerge [
-    (lib.mkIf cfg.enable (
+  config = lib.mkIf cfg.enable (lib.mkMerge [
```

Comment:
I'm a bit surprised this works?

Reply:
Why? Am I missing something? Looks like a common pattern in nixpkgs

Reply:
Looks good, I just didn't know it's possible to nest it like this!